rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/FileDownloader.java

```diff
@@ -65,6 +62,7 @@
   private MetricRegistry metrics;
   private Counter totalMBDownloaded;
   private Counter totalTimeToDownload;
+  private int generationNumber;
```

Comment:
You should not do this. FileDownloader is singleton in a system and you are keeping generationNumber for each instance here. This is also wrong as per the current logic where you keep the generationNumber of last file in the downloader and use it for all in setCacheStatus